aiohttp/web_request.py

```diff
@@ -510,10 +511,42 @@ def text(self):
         return bytes_body.decode(encoding)
 
     @asyncio.coroutine
-    def json(self, *, loads=json.loads):
-        """Return BODY as JSON."""
+    def json(self, *, loads=json.loads, allowed_types=None):
+        """ Return BODY as JSON. If the `Content-Type` is not in allowed_types,
+        then the request is rejected.
+        """
+        content_type = self.content_type
+        (mime_type, subtype, suffix, parameters) = parse_mimetype(content_type)
+        content_type = mime_type + '/' + subtype
+        if suffix != '':
+            content_type += '+' + suffix
+
+        charset = parameters.get('charset', None)
+
+        if isinstance(allowed_types, str):
+            if allowed_types == 'json':
+                if content_type != 'application/json':
+                    if not content_type.startswith('application') or \
+                       not content_type.endswith('+json'):
+                        raise HTTPNotAcceptable()
+            elif allowed_types != content_type:
+                raise HTTPNotAcceptable()
+        elif isinstance(allowed_types, set):
```

Comment:
We should keep code of aiohttp simple and clear. Simple thing should be done in  a simple way, and difficult ones should be possible.

If an specific application needs to handle such very obscure situation, it may control content-type by it's own before decoding of json. And also, I think, situation you have provided is just an imagination, not the real one. If no, please describe deeper.